web-ng/src/app/components/drawing-tools/drawing-tools.component.spec.ts

```diff
@@ -0,0 +1,324 @@
+      ),
+    }),
+    /* acl= */ Map()
+  );
+
+  beforeEach(
+    waitForAsync(() => {
+      authServiceSpy = jasmine.createSpyObj<AuthService>('AuthService', [
+        'canUserAddPointsToLayer',
+      ]);
+      authServiceSpy.canUserAddPointsToLayer.and.returnValue(true);
+
+      mockDisabled$ = new BehaviorSubject<boolean>(false);
+      mockEditMode$ = new BehaviorSubject<EditMode>(EditMode.None);
+      drawingToolsServiceSpy = jasmine.createSpyObj<DrawingToolsService>(
+        'DrawingToolsService',
+        ['getDisabled$', 'getEditMode$', 'setSelectedLayerId', 'setEditMode']
+      );
+      drawingToolsServiceSpy.getDisabled$.and.returnValue(mockDisabled$);
+      drawingToolsServiceSpy.getEditMode$.and.returnValue(mockEditMode$);
+
+      navigationServiceSpy = jasmine.createSpyObj<NavigationService>(
+        'NavigationService',
+        ['getObservationId$']
+      );
+      mockObservationId$ = new BehaviorSubject<string | null>(null);
+      navigationServiceSpy.getObservationId$.and.returnValue(
+        mockObservationId$
+      );
+
+      projectServiceSpy = jasmine.createSpyObj<ProjectService>(
+        'ProjectService',
+        ['getActiveProject$']
+      );
+      projectServiceSpy.getActiveProject$.and.returnValue(
+        of<Project>(mockProject)
+      );
+
+      TestBed.configureTestingModule({
+        imports: [DrawingToolsModule, BrowserAnimationsModule],
+        declarations: [DrawingToolsComponent],
+        providers: [
+          { provide: AuthService, useValue: authServiceSpy },
+          { provide: DrawingToolsService, useValue: drawingToolsServiceSpy },
+          { provide: NavigationService, useValue: navigationServiceSpy },
+          { provide: ProjectService, useValue: projectServiceSpy },
+        ],
+      }).compileComponents();
+    })
+  );
+
+  function resetFixture() {
+    fixture = TestBed.createComponent(DrawingToolsComponent);
+    fixture.detectChanges();
+  }
+
+  function assertElementSrcColor(element: Element, color: string) {
+    expect(
+      atob(element.getAttribute('src')!.slice(urlPrefix.length))
+    ).toContain(color);
+  }
+
+  beforeEach(() => {
+    resetFixture();
+  });
+
+  afterEach(() => {
+    fixture.destroy();
+  });
+
+  it('button group is enabled by default', () => {
+    const buttonGroup = fixture.debugElement.query(By.css('#button-group'))
+      .nativeElement;
+    expect(buttonGroup.getAttribute('ng-reflect-disabled')).toEqual('false');
+  });
+
+  it('button group is disabled when an observation is selected', fakeAsync(() => {
+    mockObservationId$.next('oid1');
+    tick();
+    // wait for async pipe to reflect
+    fixture.detectChanges();
+
+    const buttonGroup = fixture.debugElement.query(By.css('#button-group'))
+      .nativeElement;
+    expect(buttonGroup.getAttribute('ng-reflect-disabled')).toEqual('true');
+  }));
+
+  it('button group is  disabled when drawing tools service disables it', fakeAsync(() => {
```

Comment:
Consider wrapping related tests in `describe('...', () => { .. })` instead of repeating e.g. `button group`.

Reply:
Sg, grouped.